build/source/dshare/outpt_stat.f90

```diff
@@ -53,66 +53,74 @@ subroutine allocStat(meta,stat,err,message)
  integer(i4b)               :: iVar     ! index var_info array 
  integer(i4b)               :: iGRU     ! loop through GRUs
  integer(i4b)               :: iHRU     ! loop through HRUs
+ integer(i4b)               :: nGRU     ! number of GRUs
 
  ! initialize error control
  err=0; message='allocStat/'
 
- ! loop through grus
- do iGRU = 1,size(gru_struc)
 
-  ! take different action ndepending on whether the type has HRUs
-  select type(stat)   
-   type is (gru_hru_intVec)
-    ! (1) allocate the GRU level structure
-    allocate(stat%gru(size(gru_struc))                           ,stat=err)
-    if (err.ne.0) then; message=trim(message)//'GUR allocate error'; return; endif;
+ ! take different action depending on whether the type has HRUs
+ nGRU = size(gru_struc)
+ select type(stat)   
+  type is (gru_hru_intVec)
+   ! (1) allocate the GRU level structure
+   allocate(stat%gru(nGRU)                           ,stat=err,errmsg=cmessage)
+   if (err.ne.0) then; message=trim(message)//'gru_hru_intVec: GRU allocate error/'//trim(cmessage); return; endif;       
+   ! loop through grus
```

Comment:
moved GRU loop to avoid repeating `allocate(stat%gru(nGRU)` for multiple GRU cases